pkg/apis/numaplane/v1alpha1/shared_types.go

```diff
@@ -73,3 +74,20 @@ type ProgressiveStrategy struct {
 	// if ForcePromote is set, assessment will be skipped and Progressive upgrade will succeed
 	ForcePromote bool `json:"forcePromote,omitempty"`
 }
+
+// Rider defines a resource that can be deployed along with the primary child of a rollout
+type Rider struct {
+	// if set, perform progressive rollout when it changes
+	Progressive bool `json:"progressive,omitempty"`
+
+	// Definition of resource that is deployed with primary child
+	Definition runtime.RawExtension `json:"definition,omitempty"`
+}
+
+// PipelineRider defines a resource that can be deployed along with the primary child of a PipelineRollout
+type PipelineRider struct {
+	Rider `json:",inline"`
+
+	// if set, rider resource should be created for each vertex of the pipeline
+	PerVertex bool `json:"perVertex,omitempty"`
+}
```

Comment:
can you move this one into pipelinerollout_types.go since it's not shared?

Reply:
Got it, good idea